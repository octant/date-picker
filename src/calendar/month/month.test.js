/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-glamor-react'
import sinon from 'sinon'
import Enzyme, {mount} from 'enzyme'
import Adaptor from 'enzyme-adapter-react-16'

import Month from './'

Enzyme.configure({adapter: new Adaptor()})
expect.addSnapshotSerializer(serializer)

describe('<Month /> appearance', () => {
  test('displays the same', () => {
    const component = renderer.create(
      <Month startDate={new Date(2018, 5, 1)} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('<Month /> functionality', () => {
  test('displays a Grid', () => {
    const month = mount(<Month startDate={new Date(2018, 2, 25)} />)

    expect(month.find('Grid').length).toBe(1)
  })

  test('displays correct dates', () => {
    const month = mount(<Month startDate={new Date(2018, 3, 1)} />)

    expect(month.find('Item').last().text()).toBe('12')
  })

  test('sets an "id" property on each item')

  test('defaults to current month if a "selected" prop is not passed')

  test('loads the next month when next is clicked')

  test('loads the previous month when previous is clicked')

  test('the selected item is set when a Item is clicked')

  test('has 42 instances of Item')

  test('calls the clickMethod when a Item is clicked')

  test('all instances of Item are clickable')
})
