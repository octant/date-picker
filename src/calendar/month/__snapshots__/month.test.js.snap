// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Month /> appearance displays the same 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  width: 17.5em;
}

.css-1,
[data-css-1] {
  position: relative;
  box-sizing: border-box;
  float: left;
  margin: 0em;
  text-align: center;
  background-color: orangered;
  color: white;
  width: 2.5em;
  height: 2.5em;
  line-height: 2.5em;
}

.css-2,
[data-css-2] {
  position: relative;
  box-sizing: border-box;
  float: left;
  margin: 0em;
  text-align: center;
  background-color: orangered;
  color: #BBB;
  width: 2.5em;
  height: 2.5em;
  line-height: 2.5em;
}

<div
  className="css-0"
>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Su
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Mo
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Tu
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    We
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Th
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Fr
  </div>
  <div
    className="css-1"
    onClick={[Function]}
  >
    Sa
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    14
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    15
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    16
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    17
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    18
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    19
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    21
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    22
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    23
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    24
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    25
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    26
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    27
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    28
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    29
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    30
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="css-2"
    onClick={[Function]}
  >
    12
  </div>
</div>
`;
